zrml/court/src/lib.rs

```diff
@@ -104,6 +113,39 @@ mod pallet {
     where
         T: Config,
     {
+        // Returns an unique random subset of `jurors` with length `len`.
+        //
+        // If `len` is greater than the length of `jurors`, then `len` will be capped.
+        pub(crate) fn random_jurors<'a, 'b, R>(
+            jurors: &'a [(T::AccountId, Juror<BalanceOf<T>>)],
+            len: usize,
+            rng: &mut R,
+        ) -> ArrayVec<&'b (T::AccountId, Juror<BalanceOf<T>>), MAX_RANDOM_JURORS>
+        where
+            R: RngCore,
+            'a: 'b,
+        {
+            let actual_len = jurors.len().min(len);
+            jurors.choose_multiple(rng, actual_len).collect()
+        }
+
+        // Returns a pseudo random number generator implementation based on the seed
+        // provided by the `Config::Random` type and the `JurorsSelectionNonce` storage.
+        pub(crate) fn rng() -> impl RngCore {
+            let nonce = <JurorsSelectionNonce<T>>::mutate(|n| {
+                let rslt = *n;
+                *n = n.wrapping_add(1);
+                rslt
+            });
+            let mut rslt: u64 = nonce;
+            // https://github.com/paritytech/substrate/issues/8312
+            let (random_hash, _) = T::Random::random(b"zrml-court");
+            for byte in random_hash.as_ref().iter().copied() {
+                rslt = rslt.wrapping_add(byte.into());
+            }
+            StdRng::seed_from_u64(rslt)
+        }
```

Comment:
Thanks Caio for incorporating all the suggestions. The usage of the randomness and distribution of jurors throughout time should be increased significantly now. I have one last remark.

**TL;DR**
Adding the bytes reduces the entropy significantly. The best way to keep as much entropy as possible is to just pass the raw byte array that is returned from the ``T::Random::random`` function. Alternatively, concatenating the bytes to a number does also maintain entropy, the only drawback being that we don't have an integer type which can contain 32 bytes of data. The following solution maintains the initial entropy (up to 32 bytes, the most common length used for the `Hash` type) and passes it to ``StdRng``:
```suggestion
        pub(crate) fn rng() -> impl RngCore {
            let nonce = <JurorsSelectionNonce<T>>::mutate(|n| {
                let rslt = *n;
                *n = n.wrapping_add(1);
                rslt
            });
            // https://github.com/paritytech/substrate/issues/8312
            let (random_hash, _) = T::Random::random(&nonce.to_le_bytes());
            let mut seed: [u8; 32] = [0; 32];

            for (idx, byte) in random_hash.as_ref().iter().enumerate() {
                if idx == seed.len() {
                    break;
                } 

                seed[idx] = *byte;
            }

            StdRng::from_seed(seed)
        }
```

**Why is entropy reduced during addition?**
To understand this, we will use the definition of Entropy (from information theory) by Claude Shannon:
![Entropy](https://wikimedia.org/api/rest_v1/media/math/render/svg/bfe3616dee43f6287d4a4e2a557de8d48ad24926)

Imagine you have one fair 6-sided dice. The probability of each number to appear after a fair throw of the dice is 1/6. Using this formula, we get the result `H(X) = -log2(1/6) = 2.584962501 Sh, in this case 2.584962501 bits`.

Now imagine you have two fair 6-sided dices. We throw the dices and concatenate the results, i.e. `result = d1 || d2`. What is the entropy? Concatenating leads to 36 different possible outcomes, each with a probability of 1/36, because we can reach every 36 outcomes only by one combination of both dices. This leads to an entropy of:
`H(X) = -log2(1/36) = 5.169925001 bits`.

Now imagine we use the add operation to calculate the result, i.e `result = d1 + d2`. When you play that through, you'll see that there is only one option for the number 2 and 12, but 6 options for the number 7 for example. Obviously the total possible number is lower and also the distribution is uneven.
What you get is this distribution of probabilities for the potential outcomes:
![image](https://user-images.githubusercontent.com/15378334/127147825-b6b50ffa-15eb-44a0-a574-90cec1bbd83b.png)
Now if we apply Shannons formula for entropy in this scenario, we will get (shortened, because the formula is quite long):
`H(X) ≈ 3.27 bits`.

Clearly we reduced the entropy by using the add operation. Therefore we also reduced the quality of the random numbers the RNGs can calculate and consequently also the distributions of the jurors.

Reply:
Great explanation! Thanks!

Reply:
Thank you. Beautiful way to realize the byte copy you implemented there!